x/auctions/keeper/genesis_test.go

```diff
@@ -0,0 +1,168 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	testkeyring "github.com/evmos/evmos/v19/testutil/integration/evmos/keyring"
+	testnetwork "github.com/evmos/evmos/v19/testutil/integration/evmos/network"
+	testutiltx "github.com/evmos/evmos/v19/testutil/tx"
+	"github.com/evmos/evmos/v19/utils"
+	"github.com/evmos/evmos/v19/x/auctions/types"
+)
+
+func TestInitGenesis(t *testing.T) {
+	// Define var used in the mutation function
+	var (
+		existentAccAddress sdk.AccAddress
+		network            *testnetwork.UnitTestNetwork
+	)
+	moduleAccountBalance := sdk.NewInt(1)
+
+	testCases := []struct {
+		name              string
+		expPanic          bool
+		mutation          func(*types.GenesisState)
+		fundModuleAccount bool
+		postCheck         func()
+	}{
+		{
+			name:              "valid default",
+			expPanic:          false,
+			mutation:          func(_ *types.GenesisState) {},
+			fundModuleAccount: true,
+			postCheck: func() {
+				genesis := types.DefaultGenesisState()
+				ctx := network.GetContext()
+
+				params := network.App.AuctionsKeeper.GetParams(ctx)
+				assert.Equal(t, genesis.Params, params, "expected different params")
+				bid := network.App.AuctionsKeeper.GetHighestBid(ctx)
+				assert.Equal(t, genesis.Bid, bid, "expected a different bid")
+				round := network.App.AuctionsKeeper.GetRound(ctx)
+				assert.Equal(t, genesis.Round, round, "expected a different round")
+			},
+		},
+		{
+			name:     "valid with non empty bidder",
+			expPanic: false,
+			mutation: func(genesis *types.GenesisState) {
+				genesis.Bid.Sender = existentAccAddress.String()
+				genesis.Bid.BidValue.Amount = moduleAccountBalance
+			},
+			fundModuleAccount: true,
+			postCheck: func() {
+				genesis := types.DefaultGenesisState()
+				ctx := network.GetContext()
+
+				params := network.App.AuctionsKeeper.GetParams(ctx)
+				assert.Equal(t, genesis.Params, params, "expected different params")
+				bid := network.App.AuctionsKeeper.GetHighestBid(ctx)
+				expBid := types.Bid{
+					Sender:   existentAccAddress.String(),
+					BidValue: sdk.NewCoin(utils.BaseDenom, moduleAccountBalance),
+				}
+				assert.Equal(t, expBid, bid, "expected a different bid")
+				round := network.App.AuctionsKeeper.GetRound(ctx)
+				assert.Equal(t, genesis.Round, round, "expected a different round")
+			},
+		},
+		{
+			name:     "invalid non enough balance on auctions module",
+			expPanic: true,
+			mutation: func(genesis *types.GenesisState) {
+				genesis.Bid.Sender = existentAccAddress.String()
+				genesis.Bid.BidValue.Amount = sdk.NewInt(1)
+			},
+			fundModuleAccount: false,
+			postCheck:         func() {},
+		},
```

Comment:
On the implementation side there is no assumption on this. When tokens are burned it is used the bid amount instead of the entire balance:
https://github.com/evmos/evmos/blob/a0ef88bf611489604c86e2005e1b68e4bbce7478/x/auctions/keeper/hooks.go#L49-L53